Micro reactor with hydrogen redistribution Refs #308 (#317)
idaholab/virtual_test_bed: doc/content/microreactors/hpmr_h2/simba_model.md

# HPMR-H$_2$ Reactor Model

*Contact: Stefano Terlizzi, stefano.terlizzi\@inl.gov, Vincent Labour&#233;, vincent.laboure\@inl.gov*

> *Model summarized, documented, and uploaded by Nathan Grund*

Reviewer: resolved

*Model link: [Direwolf Steady State Model](https://github.com/idaholab/virtual_test_bed/tree/devel/microreactors/hpmr/steady)*

> !tag name=HPMR_H2 Direwolf Steady State Model pairs=reactor_type:MR

Reviewer: resolved

                       reactor:HPMR_H2
                       geometry:core
                       simulation_type:core_multiphysics
                       input_features:multiapps
                       code_used:DireWolf
                       computing_needs:HPC

>                        fiscal_year:2023

Reviewer: resolved

## Mesh

> The reactor module in MOOSE [!citep](MOOSEReactorModule) was used to generate three different meshes for: (a) the Discontinuous Finite Elmenet (DFEM) discrete ordinates (SN) neutronic solver shown in [simba_mesh] (a), one for the Bison model, shown in [simba_mesh] (b), and one for the Coarse Mesh Finite Difference (CMFD) acceleration mesh. displayed in [simba_mesh] (c). A one-twelfth radially reflected geometry was built by exploiting the problem symmetry to minimize the number of degrees of freedom and increase the speed of calculations [!citep](Terlizzi2023).

Reviewer: resolved

!media hpmr_h2/simba_mesh2.jpeg
    caption= (a) DFEM-SN mesh, (b) Bison mesh, and (c) CMFD mesh [!citep](Terlizzi2023).
    id=simba_mesh
    style=width:60%; margin-left:auto; margin-right:auto

## Cross Sections

> Serpent (v. 2.1.32) was used to generate the multigroup cross sections for the HPMR-H$_2$ problem. The ENDF/B-VIII.0 continuous energy library was utilized to leverage the $YH_x$ scattering libraries, which was then converted into an 11-group structure to perform the calculations. The group upper boundaries are reported in the [table-floating1].

Reviewer: converted by ISOXML which is inside Griffin (reads the Serpent tallies and converts them into a multigroup XS library readable by Griffin)

> Serpent (v. 2.1.32) was used to generate the multigroup cross sections for the HPMR-H$_2$ problem. The ENDF/B-VIII.0 continuous energy library was utilized to leverage the $YH_x$ scattering libraries, which was then converted into an 11-group structure to perform the calculations. The group upper boundaries are reported in the [table-floating1].

Reviewer: comment added

[simba_mpmap] shows the computational workflow for the coupled simulation. The left box includes the preliminary operations—namely, the mesh preparation and cross-section tabulations performed with Serpent (v. 2.1.32).

!media hpmr_h2/simba_mpmap.jpeg
    caption= Integrated computational workflow [!citep](Terlizzi2023).
    id=simba_mpmap
    style=width:80%; margin-left:auto; margin-right:auto

Griffin is run to simulate the neutron transport in the core. The power density, $P_d$, computed from the neutron flux, is then transferred to Bison that is utilized to solve for the temperature in the reflector, $T_r$, moderator, $T_m$, and fuel, $T_f$, and the hydrogen-yttrium stoichiometric ratio in the yttrium hydride, here denoted as $c_H$. Bison is coupled to 101 single heat pipes sub-applications that are used to compute the equivalent heat transfer coefficient, $h_{eq}$, and the secondary-side temperature, $T_{secondary}^i$, to obtain the thermal fluids' response. The updated temperature field and the hydrogen-yttrium stoichiometric ratio are then fed back into Griffin, where the macroscopic cross sections are updated based on the new operating conditions,

## Running the Model

> To run the input via the HPC, load the following modules:

Reviewer 2: ```suggestion
To run the input on the INL HPC, load the following modules:
```

Reviewer: resolved

```language=CPP
module load use.moose moose-apps direwolf
```

> Finally, to run the input, make sure you are in the correct directory, then use the run command below. The input being ran will be the neutronics file.

Reviewer 2: no you need a submission script, cant run this on the head node
Maybe add a line for an interactive session?

Reviewer: line for interactive session added

```language=CPP
mpirun -np 192 dire_wolf-opt -i neutronics_eigenvalue.i
```

> With these commands, the run should take 20-25 minutes. The input can be also run with the BlueCRAB executable if Sockeye executable is available using:

Reviewer 2: ```suggestion
With these commands, the run should take 20-25 minutes. The input can be also run with the BlueCRAB executable only you are using a version containing Sockeye with:
```

Reviewer: reworded
